elastic/kibana
# [Synthetics] Ensure port or url is never undefined, to prevent project monitor decryption errors

## Summary

Resolves: https://github.com/elastic/kibana/issues/134864

Ensures that url and port is never undefined on the monitor saved object, in order to preserve encrypted additional authenticated data. More info on why this change is necessary: https://github.com/elastic/kibana/issues/134864

Testing
---
1. Add the following configuration to your kibana.dev.yml: https://p.elstc.co/paste/6w+exglW#sKQNTeLN8dw27OB6MVRVbdocnNvNTc48qL+McBtLnz9
2. Start Kibana by running `yarn kbn bootstrap`, then `yarn start`
3. Navigate to `/app/uptime/manage-monitors`
4. Click API Keys, then generate api key. Copy the api key. 
5. Navigate to the `examples/todos` directory in the synthetics repo https://github.com/elastic/synthetics/tree/main/examples/todos
6. Ensure you have the latest version of @elastic/synthetics installed, then run `npx @elastic/synthetics push examples/todos/basic.journey.ts --auth [YOUR API KEY] --url [YOUR KIBANA URL]  --project test-project --schedule 10 --locations us_east` (You may want to comment out all but one journey in `basic.journey.ts` or `advanced.journey.ts` to ensure you don't hit fair use limits).
7. Navigate to `/app/uptime`. Wait for monitor results to come back.
8. One monitor results have been populated, navigate to `/app/uptime/manage-monitors`. Wait for the url field of your monitor to be populated. It should populate on the next monitor sync. The configuration provided sets the sync interval to 1m, so it should sync quickly after results have been indexed.
9. Once you confirmed that the url of the monitor appears in the `URL` column on `app/uptime/manage-monitors`, run the above synthetics command again `npx @elastic/synthetics push examples/todos/basic.journey.ts --auth [YOUR API KEY] --url [YOUR KIBANA URL]  --project test-project --schedule 10 --locations us_east`.
10. Ensure that you do not see the following errors in the logs
```
[ERROR][plugins.encryptedSavedObjects] Failed to decrypt "secrets" attribute: Unsupported state or unable to authenticate data
[plugins.taskManager] Task UPTIME:SyntheticsService:Sync-Saved-Monitor-Objects "UPTIME:SyntheticsService:sync-task" failed: Error: Unable to decrypt attribute "secrets"
```
